packages/webhook-utils/src/webhook-migration-runner.ts

```diff
@@ -1,59 +1,96 @@
-import { Client } from "urql";
-import { AppDetailsFragment } from "../generated/graphql";
 import { WebhookManifest } from "@saleor/app-sdk/types";
-import { getAppDetailsAndWebhooksData } from "./operations/get-app-details-and-webhooks-data";
-import { createLogger } from "@saleor/apps-shared";
-import { updateWebhooks } from "./update-webhooks";
-
-const logger = createLogger({ name: "updateScript" });
-
-interface WebhookMigrationRunnerArgs {
-  client: Client;
-  getManifests: ({
-    appDetails,
-  }: {
-    appDetails: AppDetailsFragment;
-  }) => Promise<Array<WebhookManifest>>;
-  dryRun?: boolean;
-}
+import { Client } from "urql";
+import { AppPermissionDeniedError, NetworkError, UnknownConnectionError } from "./errors";
+import {
+  AppDetails,
+  getAppDetailsAndWebhooksData,
+} from "./operations/get-app-details-and-webhooks-data";
+import {
+  SaleorInstanceDetails,
+  getSaleorInstanceDetails,
+} from "./operations/get-saleor-instance-details";
+import { Logger } from "./types";
+import { WebhookUpdater } from "./webhook-updater";
 
-export const webhookMigrationRunner = async ({
-  client,
-  getManifests,
-  dryRun,
-}: WebhookMigrationRunnerArgs) => {
-  logger.info("Getting app details and webhooks data");
+type GetManifestFunction = ({
+  appDetails,
+  instanceDetails,
+}: {
+  appDetails: AppDetails;
+  instanceDetails: SaleorInstanceDetails;
+}) => Promise<Array<WebhookManifest>>;
 
-  let appDetails: AppDetailsFragment | undefined;
+export class WebhookMigrationRunner {
+  private dryRun: boolean;
+  private logger: Logger;
+  private client: Client;
+  private getManifests: GetManifestFunction;
 
-  try {
-    appDetails = await getAppDetailsAndWebhooksData({ client });
-  } catch (e) {
-    logger.error(e, "Couldn't fetch the app details.");
-    return;
+  constructor(args: {
+    dryRun: boolean;
+    logger: Logger;
+    client: Client;
+    getManifests: GetManifestFunction;
+  }) {
+    this.dryRun = args.dryRun;
+    this.logger = args.logger;
+    this.client = args.client;
+    this.getManifests = args.getManifests;
   }
```

Comment:
this verbosite is not necessary. you can do `(private args: {...})` in constructor and it will be automatically assigned to `this`